services/rollout-service/pkg/versions/versions.go

```diff
@@ -72,6 +77,67 @@ func (v *versionClient) GetVersion(ctx context.Context, revision, environment, a
 	return 0, nil
 }
 
+type KuberpultEvent struct {
+	Environment string
+	Application string
+	Version     uint64
+}
+
+type VersionEventProcessor interface {
+	ProcessKuberpultEvent(ctx context.Context, ev KuberpultEvent)
+}
+
+func (v *versionClient) Subscribe(ctx context.Context, processor VersionEventProcessor) error {
+	ctx = auth.WriteUserToGrpcContext(ctx, RolloutServiceUser)
+outer:
+	for {
+		client, err := v.client.StreamOverview(ctx, &api.GetOverviewRequest{})
+		if err != nil {
+			logger.FromContext(ctx).Warn("overview.connect", zap.Error(err))
+			continue outer
```

Comment:
should we really retry immediately? E.g. if we can't connect for a long time, shouldn't we log as error, not warn?

Reply:
No

Reply:
can be solved in a later PR